### ISSUE TITLE
Having to change control mode back and forth to get it to work (see video)

### ISSUE THREAD

**Issue by DIYtechie** [jonkristian/alko#10]
Thanks for making this integration! It is greatly appreciated. 

One issue I face: When I want the mower to start, I have to set the control to "Working" and then to "Idle" and the back to "Working for it to start. And when i want it to stop, I have to set it to "Idle" and then "Working" and then "Idle" again to make it stop. As shown in this video: https://github.com/user-attachments/assets/6d2040d9-5f6d-4b28-b174-6f10f22706f7 (turn on sound)

How to fix this?

**Comment by jonkristian** [comment]
Thank you!

I am currently on vacation, but will take a look at this when I get back. I was thinking about improving this, because there is now a lawn mower entity in HA which we didn't have back when I first wrote the integration.

**Comment by DIYtechie** [comment]
Sounds great. Instead of the input selector, I made 3 button helpers, one for mowing, one for stopping and one for returning to dock. And then I translated HOMING to "Returning to dock", etc. 

Thanks again for making this integration. I really appreciate it and already use it a lot.

**Comment by jonkristian** [comment]
@DIYtechie Is this still an issue? 

Unfortunately the PSU of my base station is not working, so unable to test for the moment. Please let me know if you still have this issue, if not, feel free to close this.

**Comment by DIYtechie** [comment]
Hi Jon. Yes, I just tested and the issue remains when using the selector as shown in the video, but as mentioned above I found a workaround: I just made a script that starts the mower instead of using the selector. For some reason this works without having to toggle the state twice:

<img width="1167" alt="Image" src="https://github.com/user-attachments/assets/0818bcfa-b5fd-4603-8c61-81c774e4e87b" />

**Comment by jonkristian** [comment]
@DIYtechie in [v0.3.2](https://github.com/jonkristian/alko/releases/tag/v0.3.2) I've actually replaced the select with the [lawn_mower](https://www.home-assistant.io/integrations/lawn_mower/) integration. Curious to hear if this has fixed your issue.

**Comment by DIYtechie** [comment]
Hi! I just tested and I like that it uses the lawn mower integration. The buttons (lawn mower commands) work much better than the old "Control mode". One click and it works the first time:

![Image](https://github.com/user-attachments/assets/8e0b4fdc-5942-4e5f-884f-c04a9fead080)

Two issues, however

1) the Lawn Mower "activity state" is not working properly. It is much delayed compared to operation state while the mower is paused it shows "Docked" at least for a while. And while returning it states Mowing for a while.

2) The only lawn mower features available in the tile button are "start_pause" and "return to dock". Because the functioning of these (apparently) depend on the "activity state" it means I cannot pause or return to dock if the activity state appears to be Docked.

It would be nice to have Start, Pause and return to dock available in the tile card and not having them depent on the activity state (meaning they should not be greyed out just because the state might be wrong)

Another observation (not a big deal):
I deleted the entire integration and credentials to start fresh and I don't know if you have any control over the credentials flow, but I think it would be ideal if the user did not have to select a name for the credentials - since I was (again) a bit confused about what name to put in. It would also be nice to just have all 4 field available with user, password, client id and client secret in a single window since this helps make clear that client secret is not just another word for password etc.

**Comment by jonkristian** [comment]
Yes, this is most likely due to polling time, It used to be 120 seconds, I've now cut it down to 60. I don't want to be too aggressive here because I don't know how strict the rate limiting is. I will look into this tough, because it should set the new state straight away imo, could be an issue with the code... The lawn_mower integration is pretty limited and this is currently what we get from HA atm.

In regards to the auth flow, it's not a traditional auth flow because you have username/password together with these ordinary oauth tokens, which is pretty unusual and actually I had a real hard time getting those two to work combined 🗡 I don't think it's a huge issue since we have reconfigure now which will take your username/password/keys and reauth, so it will be more like a one-time thing.

I am not sure what you mean about name, if that is on the screen where you enter your keys I think it must come from HA, not sure I can do anything about that, but will keep it in mind when I go through it again, maybe it could have a description added if not removed or pre-populated.

**Comment by DIYtechie** [comment]
I should probably close this issue now since there is no longer an issue with control mode (which no longer exists).

I can open a new issue regarding activity state. I think a solution could be to simply translate the operation state into activity state (homing = returning etc) since the operation state seems to update fast enough.
